python/ray/train/v2/api/result.py

```diff
@@ -33,7 +42,90 @@ def from_path(
         path: Union[str, os.PathLike],
         storage_filesystem: Optional[pyarrow.fs.FileSystem] = None,
     ) -> "Result":
-        raise NotImplementedError("`Result.from_path` is not implemented yet.")
+        """Restore a Ray Train v2 Result from a persisted training run path.
+
+        Args:
+            path: Path to the experiment directory containing checkpoint_manager_snapshot.json
+            storage_filesystem: Optional filesystem to use for accessing the path
+
+        Returns:
+            Result object with restored checkpoints and metrics
+        """
+        fs, fs_path = get_fs_and_path(str(path), storage_filesystem)
+
+        # Validate that the experiment directory exists
+        if not _exists_at_fs_path(fs, fs_path):
+            raise RuntimeError(f"Experiment folder {fs_path} doesn't exist!")
+
+        # Remove trailing slashes to handle paths correctly
+        # os.path.basename() returns empty string for paths with trailing slashes
+        fs_path = fs_path.rstrip("/")
+        storage_path, experiment_dir_name = os.path.dirname(fs_path), os.path.basename(
+            fs_path
+        )
+
+        storage_context = StorageContext(
+            storage_path=storage_path,
+            experiment_dir_name=experiment_dir_name,
+            storage_filesystem=fs,
+        )
+
+        # Validate that the checkpoint manager snapshot file exists
+        if not _exists_at_fs_path(
+            storage_context.storage_filesystem,
+            storage_context.checkpoint_manager_snapshot_path,
+        ):
+            raise RuntimeError(
+                f"Failed to restore the Result object: "
+                f"{CHECKPOINT_MANAGER_SNAPSHOT_FILENAME} doesn't exist in the "
+                f"experiment folder!"
+            )
+
+        checkpoint_manager = CheckpointManager(
+            storage_context=storage_context,
+            checkpoint_config=CheckpointConfig(),
```

Comment:
```suggestion
            # NOTE: Pass a dummy checkpoint config since the checkpoint
            # manager is only used for loading the checkpoint info, and
            # we will not save further checkpoints.
            checkpoint_config=CheckpointConfig(),
```